TELCODOCS-2137#Modify generic update advice for Kube API removals in Day 2 Operations (#92256)
openshift/openshift-docs: snippets/acknowledge-the-update.adoc

:_mod-docs-content-type: SNIPPET
[IMPORTANT]
====

> Before you acknowledge the update, verify that there are no Kubernetes APIs in use that are removed in the version you are updating to.

Reviewer: ```suggestion
Before you acknowledge the update, verify that you are not using any of the Kubernetes APIs that are removed from the version you are updating to.
```

For example, in {product-title} 4.17, there are no API removals.
See "Kubernetes API removals" for more information.
====

.Prerequisites

* You have verified that APIs for all of the applications running on your cluster are compatible with the next Y-stream release of {product-title}.

> For more details about compatibility, see "Verifying cluster API versions between update versions".

Reviewer: I might have missed it, but I don't see a link in the additional resources to "Verifying cluster API versions between update versions".

Reviewer 2: Good catch, I added them now!

.Procedure

> . Perform the administrative acknowledgment to start the cluster update.

Reviewer: ```suggestion
. Complete the administrative acknowledgment to start the cluster update by entering the following command:
```
IBM Style advises to avoid "perform": https://www.ibm.com/docs/en/ibm-style?topic=word-usage#perform
OCP docs advise to introduce a command with explicit guidance to "run" or "enter" the command: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#procedure-step-that-introduces-a-command

+
--
[source,terminal]
----
$ oc adm upgrade
----

> If the cluster update does not complete successfully, more detail about the update failure is provided under the `Reason` and `Message` sections.

Reviewer: ```suggestion
If the cluster update does not complete successfully, more details about the update failure are provided in the `Reason` and `Message` sections.
```

  Reason: PreconditionChecks
  Message: Preconditions failed for payload loaded version="4.16.34" image="quay.io/openshift-release-dev/ocp-release@sha256:41bb08c560f6db5039ccdf242e590e8b23049b5eb31e1c4f6021d1d520b353b8": Precondition "ClusterVersionUpgradeable" failed because of "MultipleReasons": Cluster should not be upgraded between minor versions for multiple reasons: AdminAckRequired,ResourceDeletesInProgress
  * Kubernetes 1.29 and therefore OpenShift 4.16 remove several APIs which require admin consideration. Please see the knowledge article https://access.redhat.com/articles/7031404 for details and instructions.
  * Cluster minor level upgrades are not allowed while resource deletions are in progress; resources=PrometheusRule "openshift-kube-apiserver/kube-apiserver-recording-rules"

Upstream is unset, so the cluster will use an appropriate default.
Channel: eus-4.16 (available channels: candidate-4.15, candidate-4.16, eus-4.16, fast-4.15, fast-4.16, stable-4.15, stable-4.16)

Recommended updates:

  VERSION     IMAGE
  4.16.34     quay.io/openshift-release-dev/ocp-release@sha256:41bb08c560f6db5039ccdf242e590e8b23049b5eb31e1c4f6021d1d520b353b8
----

[NOTE]
====

> In this example, a linked Knowledge Base article (link:https://access.redhat.com/articles/7031404[Preparing to upgrade to {product-title} 4.16]) provides more detail about verifying API compatibility between releases.

Reviewer: ```suggestion
In this example, a linked Red{nbsp}Hat Knowledgebase article (link:https://access.redhat.com/articles/7031404[Preparing to upgrade to {product-title} 4.16]) provides more detail about verifying API compatibility between releases.
```

====
--

.Verification

> * Verify the update.

Reviewer: ```suggestion
* Verify the update by entering the following command:
```

> Run the following command:

Reviewer: ```suggestion

```
